### DISCUSSION TITLE
curl ssl error during installation of certain formulae (MacOS 10.11)

### DISCUSSION THREAD

**Discussion by tomasi-o** [Homebrew/discussions#472] [Everyday usage]
I know MacOS 10.11 is a rather old OS, but so is my iMac, and for some reasons I want to stick with it. When trying to install a formula like ffmpeg using homebrew (I have to use `brew install --build-rom-source ffmpeg`), installation terminates with the following error
```
==> Upgrading gnu-getopt 1.1.6 -> 2.36.1
==> Downloading https://www.kernel.org/pub/linux/utils/util-linux/v2.36/util-linux-2.36.1.tar.xz
######################################################################## 100.0%
curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
Error: Failed to download resource "gnu-getopt"
Download failed: https://www.kernel.org/pub/linux/utils/util-linux/v2.36/util-linux-2.36.1.tar.xz 
```

Does anybody know how to instruct curl not to seek for valid ssl-certificates while homebrew is using to download the necessary parts for building a formula? I searched through the curl and libcurl documentation but I am not skilled enough to understand how to do that. 
TIA for your help.

**Comment by carlocab** [comment]
You could try
```sh
echo "--insecure" >> ~/.curlrc
export HOMEBREW_CURLRC=1
```

**Reply by SMillerDev** [reply]
That would make you vulnerable to MitM attacks whenever you use a program that uses curl (which is anything that connects to the internet).

**Reply by carlocab** [reply]
Yes, it would. Delete `~/.curlrc` as soon as you’re able to install the problematic formula.

**Comment by tomasi-o** [comment]
Thank you both for your help! As I ran into multiple errors while brew tried to install the dependencies of ffmpeg, I gave up. Is running sth. like `brew cleanup` and/or `brew doctor` sufficient to clean my brew installation of all the files which where installed in this process?

**Reply by SMillerDev** [reply]
You can run the uninstall script if you want to get rid of homebrew, or run `brew autoremove` to remove all unused dependencies.

**Reply by tomasi-o** [reply]
Thank you again @SMillerDev, `brew autoremove` was exactly what I was looking for! Saved me a lot of searching.
